entitlement/src/test/java/org/killbill/billing/entitlement/api/TestDefaultEntitlement.java

```diff
@@ -466,4 +472,83 @@ public void testEntitlementChangePlanOnPendingEntitlement() throws AccountApiExc
         assertEquals(entitlement1.getLastActiveProduct().getName(), "Pistol");
 
     }
+    
+    @Test(groups = "slow")
+    public void testChangePlanWithDateTime() throws AccountApiException, EntitlementApiException, SubscriptionApiException {
+        final DateTime initialDateTime = new DateTime(2013, 8, 7, 10, 30);
+        clock.setTime(initialDateTime);
+    	
+        final Account account = createAccount(getAccountData(7));
+
+        final PlanPhaseSpecifier planPhaseSpecifier = new PlanPhaseSpecifier("Shotgun", BillingPeriod.MONTHLY, PriceListSet.DEFAULT_PRICELIST_NAME, null);
+
+        // Create entitlement and check each field
+        testListener.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK);
+        final UUID entitlementId = entitlementApi.createBaseEntitlement(account.getId(), new DefaultEntitlementSpecifier(planPhaseSpecifier), account.getExternalKey(), null, null, false, true, Collections.emptyList(), callContext);
+        assertListenerStatus();
+        final Entitlement entitlement = entitlementApi.getEntitlementForId(entitlementId, callContext);
+        assertEquals(entitlement.getState(), EntitlementState.ACTIVE);
+        assertEquals(entitlement.getLastActiveProduct().getName(), "Shotgun");
+
+        testListener.pushExpectedEvents(NextEvent.CHANGE);
+        final DateTime changeDateTime = new DateTime(2013, 8, 12, 11, 15); 
+        clock.setTime(changeDateTime);
```

Comment:
It might be interesting to set the clock to a time slightly after  `changeDateTime` (e.g 5' after) to ensure that the value being used is indeed your `changeDateTime` (and not `now`, which currently has the same value)

Reply:
Fixed